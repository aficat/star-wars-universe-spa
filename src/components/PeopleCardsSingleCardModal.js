import React, { Component } from 'react'
import { Modal } from 'semantic-ui-react';
import { Typography } from '@material-ui/core';
import { fetchHomeworld, fetchSpecies, fetchFilms, fetchStarships, fetchVehicles } from '../actions/personActions';
import { connect } from 'react-redux';

class PeopleCardsSingleCardModal extends Component {
    render() {
        const {
            name,
            height,
            mass,
            gender,
            eyeColor,
            skinColor,
            hairColor,
            birthYear,
            createdDate,
            editedDate
        } = this.props.person;
        const {
            homeworld,
            species,
            vehicles,
            starships,
            films
        } = this.props;
        const created = new Date(createdDate);
        const edited = new Date(editedDate);
        return (
            <Modal
                size={"mini"}
                open={this.props.open}
                onClose={this.props.handleClose}
                closeIcon
            >
                <Modal.Header>
                    {name}
                </Modal.Header>
                <Modal.Content scrolling>
                    <Typography
                        variant="body2"
                        component="p"
                    >
                        Gender: {gender}<br />
                        Birth Year: {birthYear}<br />
                        <br />
                        Height: {height}<br />
                        Mass: {mass}<br />
                        Eye Color: {eyeColor} <br />
                        Skin Color: {skinColor} <br />
                        Hair Color: {hairColor} <br />
                        <br />
                        Homeworld: {homeworld} <br />
                        Species: {species.length === 0 ?
                            "Unknown"
                            : (species.join(", "))} <br />
                        <br />
                        Vehicles:  {vehicles.length === 0 ?
                            "None"
                            : (vehicles.join(", "))} <br />
                            Starships:  {starships.length === 0 ?
                            "None"
                            : (starships.join(", "))} <br />
                        <br />
                            Films:  {films.length === 0 ?
                            "None"
                            : (films.join(", "))} <br />
                        <br />
                        Created on {created.toDateString()}<br />
                        Last edited on {edited.toDateString()}
                    </Typography>
                </Modal.Content>
            </Modal>
        )
    }
}

const mapStateToProps = state => ({
    homeworld: state.person.homeworld,
    species: state.person.species,
    films: state.person.films,
    starships: state.person.starships,
    vehicles: state.person.vehicles
});
export default connect(mapStateToProps,
    {
        fetchHomeworld,
        fetchSpecies,
        fetchFilms,
        fetchStarships,
        fetchVehicles
    })(PeopleCardsSingleCardModal);
