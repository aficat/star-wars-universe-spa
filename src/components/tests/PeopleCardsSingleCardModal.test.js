import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json'
import PeopleCardsSingleCardModal from '../PeopleCardsSingleCardModal';
import store from '../../store';

describe('PeopleCardsSingleCardModal component', () => {

    it('renders without crashing given the required props', () => {
        const props = {
            dispatch: jest.fn(),
            homeworld: "",
            species: [],
            films: [],
            starships: [],
            vehicles: [],
        }
        const wrapper = shallow(<PeopleCardsSingleCardModal {...props} store={store} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

});