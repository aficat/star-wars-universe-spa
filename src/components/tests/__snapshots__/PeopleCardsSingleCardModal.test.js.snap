// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleCardsSingleCardModal component renders without crashing given the required props 1`] = `
<ContextProvider
  value={null}
>
  <PeopleCardsSingleCardModal
    dispatch={[MockFunction]}
    fetchFilms={[Function]}
    fetchHomeworld={[Function]}
    fetchSpecies={[Function]}
    fetchStarships={[Function]}
    fetchVehicles={[Function]}
    films={Array []}
    homeworld=""
    species={Array []}
    starships={Array []}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    vehicles={Array []}
  />
</ContextProvider>
`;
